pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_service.go

```diff
@@ -270,3 +270,38 @@ func getLabels() map[string]string {
 func isShootNamespace(namespace string) bool {
 	return strings.HasPrefix(namespace, v1beta1constants.TechnicalIDPrefix)
 }
+
+type kubeAPIServerService struct {
+	client    client.Client
+	namespace string
+}
+
+// NewKubeAPIServerService creates a new instance of Deployer for the service pointing to kubernetes.default.svc.cluster.local.
+func NewKubeAPIServerService(c client.Client, namespace string) component.Deployer {
+	return &kubeAPIServerService{
+		client:    c,
+		namespace: namespace,
+	}
+}
+
+func (ks *kubeAPIServerService) Deploy(ctx context.Context) error {
+	svc := ks.emptyService()
+
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, ks.client, svc, func() error {
+		svc.Labels = utils.MergeStringMaps(svc.Labels, getLabels())
+		svc.Spec = corev1.ServiceSpec{
+			Type:         corev1.ServiceTypeExternalName,
+			ExternalName: "kubernetes.default.svc.cluster.local",
```

Comment:
Maybe you can reuse the logic of https://github.com/gardener/gardener/blob/a36c6731581a146920f4bcd31a14528fef5a2a71/pkg/utils/kubernetes/service.go#L24 by introducing another function like `FQDNForService`.